PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,326 @@
+
+    /// All product identifiers purchases by the user regardless of expiration.
+    @objc public var allPurchasedProductIdentifiers: Set<String> {
+        return self.nonConsumablePurchases.union(expirationDatesByProduct.keys)
+    }
+
+    /// Returns the latest expiration date of all products, nil if there are none
+    @objc public var latestExpirationDate: Date? {
+        let mostRecentDate = self.expirationDatesByProduct
+            .values
+            .compactMap({$0})
+            .max(by: { $0.timeIntervalSinceReferenceDate < $1.timeIntervalSinceReferenceDate })
+
+        return mostRecentDate
+    }
+
+    /// Returns all product IDs of the non-subscription purchases a user has made.
+    @available(*, deprecated, message: "use nonSubscriptionTransactions")
+    @objc public let nonConsumablePurchases: Set<String>
+
+    /**
+    Returns all the non-subscription purchases a user has made.
+    The purchases are ordered by purchase date in ascending order.
+     */
+    @objc public let nonSubscriptionTransactions: [Transaction]
+
+    /**
+     Returns the fetch date of this Purchaser info.
+     @note Can be nil if was cached before we added this
+     */
+    @objc public let requestDate: Date
+
+    /// The date this user was first seen in RevenueCat.
+    @objc public let firstSeen: Date
+
+    /// The original App User Id recorded for this user.
+    @objc public let originalAppUserId: String
+
+    /** URL to manage the active subscription of the user.
+     If this user has an active iOS subscription, this will point to the App Store,
+     if the user has an active Play Store subscription it will point there.
+     If there are no active subscriptions it will be null.
+     If there are multiple for different platforms, it will point to the App Store
+     */
+    @objc public let managementURL: URL?
+
+    /**
+    Returns the purchase date for the version of the application when the user bought the app.
+    Use this for grandfathering users when migrating to subscriptions.
+
+    @note This can be nil, see -[RCPurchases restoreTransactionsForAppStore:]
+     */
+    @objc public let originalPurchaseDate: Date?
+
+    /**
+    Returns the build number (in iOS) or the marketing version (in macOS) for the version of the application when the user bought the app.
+    This corresponds to the value of CFBundleVersion (in iOS) or CFBundleShortVersionString (in macOS) in the Info.plist file when the purchase was originally made.
+    Use this for grandfathering users when migrating to subscriptions.
+
+     
+     @note This can be nil, see -[RCPurchases restoreTransactionsForAppStore:]
+     */
+    @objc public let originalApplicationVersion: String?
+
+    // TODO after migration make this internal
+    @objc public let schemaVersion: String?
+
+    private let expirationDatesByProduct: [String: Date?]
+
+    private let purchaseDatesByProduct: [String: Date?]
```

Comment:
perhaps we should be more explicit here with the fact that the keys aren't products, but rather product identifiers? 
i.e.: `expirationDatesByProductId`